Add a section for overriding APP_* token providers (#1578)
nestjs/docs.nestjs.com: content/fundamentals/unit-testing.md

> info **Hint** Keep your e2e test files inside the `e2e` directory. The testing files should have a `.e2e-spec` or `.e2e-test` suffix.

> #### Overriding globally registered providers

Reviewer: ```suggestion
#### Overriding globally registered enhancers
```

> If you have a [globally registered guard](/security/authentication#enable-authentication-globally) (or pipe, interceptor, or filter), you need to take a few more steps to override that enhancer. To recap the original registration looks like this:

Reviewer: ```suggestion
If you have a globally registered guard (or pipe, interceptor, or filter), you need to take a few more steps to override that enhancer. To recap the original registration looks like this:
```

> If you have a [globally registered guard](/security/authentication#enable-authentication-globally) (or pipe, interceptor, or filter), you need to take a few more steps to override that enhancer. To re-cap the original registration looks like this:

Reviewer 2: `recap` instead of `re-cap`

> If you have a [globally registered guard](/security/authentication#enable-authentication-globally) (or pipe, interceptor, or filter), you need to take a few more steps to override that enhancer. To re-cap the original registration looks like this:

Reviewer 2: ```suggestion
If you have a [globally registered guard](/security/authentication#enable-authentication-globally) (or pipe, interceptor, or filter), you need to take a few more steps to override that enhancer. To recap the original registration looks like this:
```

```typescript
providers: [
  {
    provide: APP_GUARD,
    useClass: JwtAuthGuard,
  },
],
```

> This is registering the guard as a "multi"-provider through the `APP_*` token. To be able to replace the JwtAuthGuard here, the registration needs to use an existing provider in this slot:

Reviewer: ```suggestion
This is registering the guard as a "multi"-provider through the `APP_*` token. To be able to replace the `JwtAuthGuard` here, the registration needs to use an existing provider in this slot:
```

```typescript
providers: [
  {
    provide: APP_GUARD,
    useExisting: JwtAuthGuard,
  },
  JwtAuthGuard,
],
```

> info **Hint** Change the `useClass` to `useExisting` to reference a registered provider instead of having Nest instantiate it behind the token.

> Now the `JwtAuthGuard` is visible to nest as a regular provider that can be overridden when creating the test module:

Reviewer 2: please use `nestjs` instead of `nest`

> Now the `JwtAuthGuard` is visible to nest as a regular provider that can be overridden when creating the test module:

Reviewer 2: `TestingModule` instead of `test module`

> Now the `JwtAuthGuard` is visible to nest as a regular provider that can be overridden when creating the test module:

Reviewer 2: ```suggestion
Now the `JwtAuthGuard` is visible to nestjs as a regular provider that can be overridden when creating the `TestingModule`:
```

> Now the `JwtAuthGuard` is visible to nest as a regular provider that can be overridden when creating the `TestingModule`:

Reviewer: ```suggestion
Now the `JwtAuthGuard` is visible to Nest as a regular provider that can be overridden when creating the `TestingModule`:
```

> info **Hint** Change the `useClass` to `useExisting` to reference a registered provider instead of having Nest instantiate it behind the token.

> Now the `JwtAuthGuard` is visible to nestjs as a regular provider that can be overridden when creating the `TestingModule`:

Reviewer 3: ```suggestion
Now the `JwtAuthGuard` is visible to nest as a regular provider that can be overridden when creating the `TestingModule`:
```

There's other places in the docs where the framework is referred to just as `nest` and it reads a bit more easily. I'd prefer to follow that if possible.
